roachtest: factor out downloadStoreDumps

tbg opened the pull request:
This is helpful for other tests that want to download them (I have one
in the pipes).

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/25772)
<!-- Reviewable:end -->

tschottdorf commented on pkg/cmd/roachtest/backup.go:
```diff
@@ -18,9 +18,22 @@ package main
+	"golang.org/x/sync/errgroup"
 )
 
+func downloadStoreDumps(ctx context.Context, c *cluster, location string, binVersion string, nodeCount int) error {
```
Eagle eyes please, @petermattis. Don't want to break this again. FWIW, I've run my in-progress test that uses this and it worked.

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/cmd/roachtest/backup.go, line 25 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/25772#-LD2yTzKD2vi5XhROBkl-r1-25:-LD2yyMU1qyvnA_eOK_U:bi0w9a2) ([raw file](https://github.com/cockroachdb/cockroach/blob/fa75cb7fd85669d54f65f87519cf4113f6603348/pkg/cmd/roachtest/backup.go#L25)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary><blockquote>

Eagle eyes please, @petermattis. Don't want to break this again. FWIW, I've run my in-progress test that uses this and it worked.
</blockquote></details>

LGTM. You can run the `backup2TB` test and verify that the download commands it is executing are correct and then kill the test.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/25772#-:-LD2z5xE5bTDjrRYrsJt:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
bors r=petermattis

Worked like a charm.

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/25772#-:-LD335PhDryr_0jTIXDa:b-aecx4v)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
bors r-

didn't see the lint failure.

tbg committed:
roachtest: factor out downloadStoreDumps

This is helpful for other tests that want to download them (I have one
in the pipes).

Release note: None

tschottdorf commented:
bors r=petermattis
